prow/github/fakegithub/fakegithub.go

```diff
@@ -64,6 +64,10 @@ type FakeClient struct {
 	Milestone    int
 	MilestoneMap map[string]int
 
+	// list of commits for each PR
+	// org/repo#number:[]commit
+	CommitMap map[string][]github.RepositoryCommit
```

Comment:
I think there is a similar *Repository* ListCommits (i.e. not bound to a PR). Should this field be updated to handle this distinction?